include/seqan3/alphabet/quality/phred.hpp

```diff
@@ -0,0 +1,155 @@
+
+namespace seqan3
+{
+
+/*!
+ * Implementation of the Illumina 1.8 standard fulfilling the quality concept. 
+ * The permitted phred score range is [0 .. 41], mapped to ascii-ordered range ['!' .. 'J'].
+ * For this standard internal and integral phred representation are both zero-based.
+ */
+struct illumina18
+{
+    //! the 3 representation types of a quality score
+    using phred_type = int8_t;
+    using integral_type = uint8_t;
+    using char_type = char;
+
+    //! internal integral value representation
+    integral_type value;
+    
+    //! projection offsets of char and integral quality score
+    static constexpr char_type offset_char{'!'};
+    static constexpr phred_type offset_phred{0};
+
+    //! implicit compatibility to inner_type
+    constexpr illumina18 & operator =(integral_type const c)
+    {
+        value = c;
+        return *this;
+    }
+
+    //! comparison operators
+    constexpr bool operator==(const illumina18 & rhs) const
+    {
+        return this->value == rhs.value;
+    }
+
+    constexpr bool operator!=(const illumina18 & rhs) const
+    {
+        return this->value != rhs.value;
+    }
+
+    constexpr bool operator<(const illumina18 & rhs) const
+    {
+        return this->value < rhs.value;
+    }
+
+    constexpr bool operator>(const illumina18 & rhs) const
+    {
+        return this->value > rhs.value;
+    }
+
+    constexpr bool operator<=(const illumina18 & rhs) const
+    {
+        return this->value <= rhs.value;
+    }
+
+    constexpr bool operator>=(const illumina18 & rhs) const
+    {
+        return this->value >= rhs.value;
+    }
+
+    //! explicit compatibility to char code of a quality score
+    explicit constexpr operator char() const
+    {
+        return to_char();
+    }
+
+    //! convert quality score to its 1-letter code
+    constexpr char_type to_char() const
+    {
+        return value + offset_char;
+    }
+
+    //! set internal value given 1-letter code
+    constexpr illumina18 from_char(char_type const c)
+    {
+        value = c - '!';
+        return *this;
+    }
+
+    //! explicit compatibility to internal integral representation
+    constexpr integral_type to_integral() const
+    {
+        return value;
+    }
+
+    //! set internal value given zero-based integer c
+    constexpr illumina18 from_integral(integral_type const c)
+    {
+        value = c;
+        return *this;
+    }
+
+    //! set internal value given Illumina 1.8 integer code p
+    constexpr illumina18 from_phred(phred_type const p)
+    {
+        value = p - offset_phred;
+        return *this;
+    }
+
+    //! get Illumina 1.8 integer code
+    constexpr phred_type to_phred() const
+    {
+        return value + offset_phred;
+    }
+
+    //! phred score range for Illumina 1.8 standard
+    static constexpr integral_type value_size{42};
+};
+
+//! assert when (internal) quality concept requirements are not met
+static_assert(quality_concept<illumina18>);
```

Comment:
One last thing: please wrap this in a #ifndef NDEBUG block! see the other alphabets for more details!